x/crs/keeper/msg_server.go

```diff
@@ -47,13 +41,20 @@ func (ms msgServer) CreateDecision(ctx context.Context, msg *crs.MsgCreateDecisi
 		return nil, fmt.Errorf("reveal duration must be greater than 0")
 	}
 
+	// check if entry fee is nil
 	if !msg.EntryFee.IsNil() {
 		err = ms.k.bankKeeper.SendCoinsFromAccountToModule(ctx, senderAddr, crs.ModuleName, sdk.NewCoins(msg.EntryFee))
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	//check that the senders balance is greater than the entry fee
+	balance := ms.k.bankKeeper.Balance(ctx, senderAddr)
+	if balance.AmountOf(msg.EntryFee.Denom).LT(msg.EntryFee.Amount) {
+		return nil, fmt.Errorf("sender does not have enough balance")
+	}
+
```

Comment:
Is this needed? The Send above will return an error if there is not enough tokens in the users' wallet